File: ts/project/.sdk/tm/csharp/utility/TransformResponse.cs
// ProjectName SDK utility: transformResponse - apply the point's response
// transform (when defined) to derive the result data.

using Voxgig.Struct;

namespace ProjectNameSdk.Util;

public static partial class SdkUtility
{
    internal static object? TransformResponseUtil(Context ctx)
    {
        var spec = ctx.Spec;
        var result = ctx.Result;
        var point = ctx.Point;

        if (spec != null)
        {
            spec.Step = "resform";
        }

        if (result == null || !result.Ok)
        {
            return null;
        }

        var transform = Helpers.ToMapAny(StructUtils.GetProp(point, "transform"));
        if (transform == null)
        {
            return null;
        }

        var resform = StructUtils.GetProp(transform, "res");
        if (resform == null)
        {
            return null;
        }

        var resdata = StructUtils.Transform(new Dictionary<string, object?>
        {
            ["ok"] = result.Ok,
            ["status"] = result.Status,
            ["statusText"] = result.StatusText,
            ["headers"] = result.Headers,
            ["body"] = result.Body,
            ["err"] = result.Err,
            ["resdata"] = result.Resdata,
            ["resmatch"] = result.Resmatch,
        }, resform);

        result.Resdata = resdata;
        return resdata;
    }
}

File: ts/project/.sdk/tm/java/utility/TransformResponse.java
package JAVAPACKAGE.utility;

import java.util.LinkedHashMap;
import java.util.Map;

import JAVAPACKAGE.core.Context;
import JAVAPACKAGE.core.Helpers;
import JAVAPACKAGE.core.Result;
import JAVAPACKAGE.utility.struct.Struct;

final class TransformResponse {

  private TransformResponse() {}

  static Object transformResponse(Context ctx) {
    Result result = ctx.result;

    if (ctx.spec != null) {
      ctx.spec.step = "resform";
    }

    if (result == null || !result.ok) {
      return null;
    }

    Map<String, Object> transform =
        Helpers.toMapAny(Struct.getprop(ctx.point, "transform"));
    if (transform == null) {
      return null;
    }

    Object resform = Struct.getprop(transform, "res", null);
    if (resform == null) {
      return null;
    }

    Map<String, Object> data = new LinkedHashMap<>();
    data.put("ok", result.ok);
    data.put("status", result.status);
    data.put("statusText", result.statusText);
    data.put("headers", result.headers);
    data.put("body", result.body);
    data.put("err", result.err);
    data.put("resdata", result.resdata);
    data.put("resmatch", result.resmatch);

    Object resdata = Struct.transform(data, resform);

    result.resdata = resdata;
    return resdata;
  }
}
